pkg/apis/core/validation/shoot.go

```diff
@@ -735,13 +735,43 @@ func ValidateWorker(worker core.Worker, fldPath *field.Path) field.ErrorList {
 		}
 	}
 
+	volumeSizeRegex, _ := regexp.Compile(`^(\d)+Gi$`)
+
 	if worker.Volume != nil {
-		volumeSizeRegex, _ := regexp.Compile(`^(\d)+Gi$`)
 		if !volumeSizeRegex.MatchString(worker.Volume.Size) {
 			allErrs = append(allErrs, field.Invalid(fldPath.Child("volume", "size"), worker.Volume.Size, fmt.Sprintf("volume size must match the regex %s", volumeSizeRegex)))
 		}
 	}
 
+	if worker.DataVolumes != nil {
+		var volumeNames = make(map[string]int)
+		for idx, volume := range worker.DataVolumes {
+			idxPath := fldPath.Child("dataVolumes").Index(idx)
+			if _, keyExist := volumeNames[*volume.Name]; keyExist {
+				volumeNames[*volume.Name]++
+				allErrs = append(allErrs, field.Invalid(idxPath.Child("name"), volume.Name, fmt.Sprintf("data volume name must be unique")))
+			} else {
+				volumeNames[*volume.Name] = 1
+			}
+			if !volumeSizeRegex.MatchString(volume.Size) {
+				allErrs = append(allErrs, field.Invalid(idxPath.Child("size"), volume.Size, fmt.Sprintf("data volume size must match the regex %s", volumeSizeRegex)))
+			}
+		}
+
+	}
+
```

Comment:
General question: Is it allowed to specify `worker.dataVolumes` if `worker.volume == nil`?

Reply:
I will disable this option. It makes sense for a user to designate a specific volume as the root volume. Most providers use either a dedicated field or some attribute to distinguish this volume.